<!DOCTYPE HTML>
<!--
	Spectral by HTML5 UP
	html5up.net | @ajlkn
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>
	<head>
		<title>Bioinf- Data Analysis Project</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
		<noscript><link rel="stylesheet" href="assets/css/noscript.css" /></noscript>
		<link rel="icon" type="image/png" sizes="16x16" href="images/favicon_io/favicon-16x16.png">
	</head>
	<body class="is-preload">

		<!-- Page Wrapper -->
			<div id="page-wrapper">

				<!-- Header -->
					<header id="header">
						<h1><a href="index.html">Home</a></h1>
						<nav id="nav">
							<ul>
								<li class="special">
									<a href="#menu" class="menuToggle"><span>Menu</span></a>
									<div id="menu">
										<ul>
											<li><a href="index.html">Home</a></li>
											<li><a href="introduction.html">Introduction</a></li>
											<li><a href="methods.html">Methods</a></li>
											<li><a href="results.html">Results</a></li>
											<li><a href="discussion.html">Discussion</a></li>
											<li><a href="conclusion.html">Conclusion & References</a></li>
											<li><a href="supplementary.html">Supplementary</a></li>
											<li><a target="popup" onclick="window.open('glossory.html','name','width=600,height=500')">Glossary</a></li>
										</ul>
									</div>
								</li>
							</ul>
						</nav>
					</header>
					<div class="sidenav">
						<li class="icon fa-regular fa-bookmark"> Conclusion &<br> References</i>
						
						<a href="conclusion.html">Conclusion</a>
						<a href="#ref">References</a>

					</div>

				<!-- Main -->
					<article id="main">

						<section class="wrapper style5">
							<div class="inner">
								
								<h2>5 Conclusion</h2>
								<p>
									In this analysis, three bioinformatics tools for the discovery of RiPP were evaluated, namely, MetaMiner, NeuRipp and FBGAN. Publicly available sequence data from the NCBI and antiSMASH databases were extracted and analysed using these tools. A wide variety of bioinformatics programmes, including AntiSMASH, Prodigal and HMMER packages, were also used for the pre-processing and analysis of the data. Results of the analysis highlight the limitations of each RiPP discovery tool and led to the discussion of the integration of currently available tools and future modifications that would allow a more comprehensive study of RiPPs. To conclude, currently available RiPP discovery tools are still limited by either our current knowledge about the biosynthetic processes of RiPP or the number of empirically validated precursor peptides. However, they have laid a solid foundation for the future development of exhaustive tools that would facilitate the identification of RiPPs in a high throughput manner or the generation of <i>de novo</i> RiPP sequences.
								</p>
								<h2>6 References</h2>
								<ol>
									<li>Bahar AA, Ren D. Antimicrobial Peptides. Pharmaceuticals. 2013 Dec;6(12):1543–75. </li>
									<li>Deslouches B, Di YP. Antimicrobial peptides with selective antitumor mechanisms: prospect for anticancer applications. Oncotarget. 2017 Jul 11;8(28):46635–51. </li>
									<li>Cao L, Gurevich A, Alexander KL, Naman CB, Leão T, Glukhov E, et al. MetaMiner: A Scalable Peptidogenomics Approach for Discovery of Ribosomal Peptide Natural Products with Blind Modifications from Microbial Communities. Cell Syst. 2019 Dec 18;9(6):600-608.e4. </li>
									<li>Gurevich A, Mikheenko A, Shlemov A, Korobeynikov A, Mohimani H, Pevzner PA. Increased diversity of peptidic natural products revealed by modification-tolerant database search of mass spectra. Nat Microbiol. 2018 Mar;3(3):319–27. </li>
									<li>M. Rowe S, R. Spring D. The role of chemical synthesis in developing RiPP antibiotics. Chem Soc Rev. 2021;50(7):4245–58. </li>
									<li>Arnison PG, Bibb MJ, Bierbaum G, Bowers AA, Bugni TS, Bulaj G, et al. Ribosomally synthesized and post-translationally modified peptide natural products: overview and recommendations for a universal nomenclature. Nat Prod Rep. 2013 Jan;30(1):108–60. </li>
									<li>Moffat AD, Santos-Aberturas J, Chandra G, Truman AW. A User Guide for the Identification of New RiPP Biosynthetic Gene Clusters Using a RiPPER-Based Workflow. In: Barreiro C, Barredo JL, editors. Antimicrobial Therapies: Methods and Protocols [Internet]. New York, NY: Springer US; 2021 [cited 2022 Jun 9]. p. 227–47. (Methods in Molecular Biology). Available from: https://doi.org/10.1007/978-1-0716-1358-0_14</li>
									<li>Blin K, Wolf T, Chevrette MG, Lu X, Schwalen CJ, Kautsar SA, et al. antiSMASH 4.0—improvements in chemistry prediction and gene cluster boundary identification. Nucleic Acids Res. 2017 Jul 3;45(Web Server issue):W36–41. </li>
									<li>Santos-Aberturas J, Chandra G, Frattaruolo L, Lacret R, Pham TH, Vior NM, et al. Uncovering the unexplored diversity of thioamidated ribosomal peptides in Actinobacteria using the RiPPER genome mining tool. Nucleic Acids Res. 2019 May 21;47(9):4624–37. </li>
									<li>Morton JT, Freed SD, Lee SW, Friedberg I. A large scale prediction of bacteriocin gene blocks suggests a wide functional spectrum for bacteriocins. BMC Bioinformatics. 2015 Nov 11;16:381. </li>
									<li>Eddy SR. Accelerated Profile HMM Searches. PLOS Comput Biol. 2011 Oct 20;7(10):e1002195. </li>
									<li>De Los Santos, C EL. NeuRiPP: Neural network identification of RiPP precursor peptides. Sci Rep. 2019 Sep 16;9(1):13406. </li>
									<li>Skinnider MA, Johnston CW, Gunabalasingam M, Merwin NJ, Kieliszek AM, MacLellan RJ, et al. Comprehensive prediction of secondary metabolite structure and biological activity from microbial genome sequences. Nat Commun. 2020 Nov 27;11(1):6058. </li>
									<li>Li J, Qu X, He X, Duan L, Wu G, Bi D, et al. ThioFinder: A Web-Based Tool for the Identification of Thiopeptide Gene Clusters in DNA Sequences. PLOS ONE. 2012 Sep 24;7(9):e45878. </li>
									<li>Tietz JI, Schwalen CJ, Patel PS, Maxson T, Blair PM, Tai HC, et al. A new genome-mining tool redefines the lasso peptide biosynthetic landscape. Nat Chem Biol. 2017 May;13(5):470–8. </li>
									<li>Gupta A, Zou J. Feedback GAN for DNA optimizes protein functions. Nat Mach Intell. 2019 Feb;1(2):105–11. </li>
									<li>Aggarwal A, Mittal M, Battineni G. Generative adversarial network: An overview of theory and applications. Int J Inf Manag Data Insights. 2021 Apr 1;1(1):100004. </li>
									<li>Hyatt D, Chen GL, LoCascio PF, Land ML, Larimer FW, Hauser LJ. Prodigal: prokaryotic gene recognition and translation initiation site identification. BMC Bioinformatics. 2010 Mar 8;11(1):119. </li>
									<li>Seyedsayamdost MR. High-throughput platform for the discovery of elicitors of silent bacterial gene clusters. Proc Natl Acad Sci. 2014 May 20;111(20):7266–71. </li>
									<li>Li W, O’Neill KR, Haft DH, DiCuccio M, Chetvernin V, Badretdin A, et al. RefSeq: expanding the Prokaryotic Genome Annotation Pipeline reach with protein family model curation. Nucleic Acids Res. 2021 Jan 8;49(D1):D1020–8. </li>
									<li>Cardoso MH, Orozco RQ, Rezende SB, Rodrigues G, Oshiro KGN, Cândido ES, et al. Computer-Aided Design of Antimicrobial Peptides: Are We Generating Effective Drug Candidates? Front Microbiol [Internet]. 2020 [cited 2022 May 18];10. Available from: <a href="https://www.frontiersin.org/article/10.3389/fmicb.2019.03097">https://www.frontiersin.org/article/10.3389/fmicb.2019.03097</a></li>
									<li>Toropova MA, Veselinović AM, Veselinović JB, Stojanović DB, Toropov AA. QSAR modeling of the antimicrobial activity of peptides as a mathematical function of a sequence of amino acids. Comput Biol Chem. 2015 Dec;59 Pt A:126–30. </li>
									<li>Hudson GA, Mitchell DA. RiPP Antibiotics: Biosynthesis and Engineering Potential. Curr Opin Microbiol. 2018 Oct;45:61–9. </li>
									
									
								</ol>

							</div>
						</section>
						</section>
						<section class="wrapper style5">
							<ul class="actions special">
								<li><a href="discussion.html" class="button primary fit">Discussion</a></li>
								<li><a href="supplementary.html" class="button primary fit">Supplementary</a></li>
							</ul>
						</section>
					</article>

				<!-- Footer -->
				<footer id="footer">
					<ul class="icons">
						<li><a href="https://www.imperial.ac.uk/life-sciences/postgraduate/masters-courses/msc-in-bioinformatics-and-theoretical-systems-biology/" target="_blank" class="icon brands fa-dribbble"><span class="label">Dribbble</span></a></li>
						<li><a href="mailto: yuting.tseng21@imperial.ac.uk" target="_blank" class="icon solid fa-envelope"><span class="label">Email</span></a></li>
					</ul>
					<ul class="copyright">
						<li>YTT16</li><li>Design: <a href="http://html5up.net">HTML5 UP</a></li>
					</ul>
				</footer>

			</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/jquery.scrollex.min.js"></script>
			<script src="assets/js/jquery.scrolly.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>

	</body>
</html>