<!DOCTYPE HTML>
<!--
	Spectral by HTML5 UP
	html5up.net | @ajlkn
	Free for personal and commercial use under the CCA 3.0 license (html5up.net/license)
-->
<html>
	<head>
		<title>Bioinf- Data Analysis Project</title>
		<meta charset="utf-8" />
		<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
		<link rel="stylesheet" href="assets/css/main.css" />
		<noscript><link rel="stylesheet" href="assets/css/noscript.css" /></noscript>
		<link rel="icon" type="image/png" sizes="16x16" href="images/favicon_io/favicon-16x16.png">
	</head>
	<body class="is-preload">

		<!-- Page Wrapper -->
			<div id="page-wrapper">

				<!-- Header -->
					<header id="header">
						<h1><a href="index.html">Home</a></h1>
						<nav id="nav">
							<ul>
								<li class="special">
									<a href="#menu" class="menuToggle"><span>Menu</span></a>
									<div id="menu">
										<ul>
											<li><a href="index.html">Home</a></li>
											<li><a href="introduction.html">Introduction</a></li>
											<li><a href="methods.html">Methods</a></li>
											<li><a href="results.html">Results</a></li>
											<li><a href="discussion.html">Discussion</a></li>
											<li><a href="conclusion.html">Conclusion & References</a></li>
											<li><a href="supplementary.html">Supplementary</a></li>
											<li><a target="popup" onclick="window.open('glossory.html','name','width=600,height=500')">Glossary</a></li>
										</ul>
									</div>
								</li>
							</ul>
						</nav>
					</header>
					<div class="sidenav">
						<li class="icon solid fa-list-ol"> Results</li>
						<a href="#ch3.1">3.1 MetaMiner</a>
						<a href="#ch3.2">3.2 NeuRiPP</a>
					</div>
				<!-- Main -->
					<article id="main">
						<!-- <header>
							<h2>Results</h2>
							<p>Aliquam ut ex ut interdum donec amet imperdiet eleifend</p>
						</header> -->
						
						<section class="wrapper style5">
							<a name="ch3.1"></a>
							<div class="inner">
								<h2>Results</h2>
								<h3>3.1 MetaMiner</h3>
								<h3>3.1.1 MetaMiner- STANDARD dataset</h3>
								<p>
									Three out of the six strains described in the paper were able to go through the genome analysis stage of the programme while the other three sequences failed at the genome analysis stage. Among the strains that were successfully analysed, none of them produced any significant match with the spectra. However, six peptides were reported in the MetaMiner paper, including 4 from <i>P. marinus</i>, 2 from <i>G. thermodenitrificans</i> and 1 from <i>B. halodurans</i>. To further investigate the matches, the peptides reported to have significant matches in the paper were manually queried against insignificant matches under the p-value threshold. One of the reported PP sequences from <i>P. marinus</i> was found to produce an insignificant match. The sequence was found to match with the same spectra file with 26 modification variations with the highest variation with a high p-value of 1.5E-03. 
								</p>
								<h3>3.1.2 MetaMiner- ACTI dataset</h3>
								<p>
									The ACTI dataset contains three different strains of bacteria. The MetaMiner program accepts input either to be a path to a directory or a single file. Analysis was performed in two ways. Firstly a path to the directory containing all three fasta files was specified as the input and secondly three sequence files were processed as individual files in three separate runs. When three sequences were processed together, there were two peptides producing lantibiotic identified in <i>S. roseosporous</i>. However, when the fasta files were processed in separate runs, <i>S. roseosporous</i> only generated one match. The peptide only identified in the first analysis was reported in the MetaMiner paper while the peptide found when the nucleotide sequence was processed individually was not reported in the paper. The two matches have close p-values of 5.10E-12 and 7.60E-12, respectively.</p>
								<h3>3.1.3 MetaMiner- SPACE dataset</h3>
								<p>
									The SPACE dataset contains a total of 21 isolated strains. This dataset is the only dataset where sequence files were directly downloadable from the FTP links provided in the paper. The analysis was first performed with all sequences processed in one single command line run and no significant matches were produced. However, when the sequences were processed as individual runs, three significant matches were generated for <i>Bacillus sp. JEM2</i> and <i>ISSFR_3F</i>. The three matches were identical with the same peptide sequence matching onto the same spectra files. Only one out of the three matches was reported in the MetaMiner paper. Additionally, a manual inspection was carried out to confirm the nucleotide sequences of  <i>Bacillus sp. JEM2</i> and <i>ISSFR_3F</i> are not identical.</p>
								<h4>Table 1. Strains of bacteria used for the evaluation of MetaMiner and results summary</h4>
								<div id="observablehq-viewof-search-a702b4ec"></div>
								<div id="observablehq-meta_results_table-a702b4ec"></div>
								<img src="images/metaminer_workflow1024_1.jpg" alt="" />
								<h3>3.1.4 Wrapper structure</h3>
								<p>
									The organisation of the wrapper of the MetaMiner pipeline was then analysed. The main metaminer.py script controls the main flow of the process and in this script,  input files flow through three main stages sequentially. The stages include the processing of the sequence, spectra files and genome analysis (fig. 6). During the sequence processing stage, one of three scripts is called depending on the format of the input sequence file (fasta, antiSMASH output or BOA output). PSM are collected in the spectra processing stage. When there are multiple spectra or sequence files, the run_parallel function is called to allow simultaneous processing. </p>
								
								<img src="images/fig_6.jpg" width="100%">
								<a name="ch3.2"></a>
								<h6><b>Figure 6. A high-level workflow of the MetaMiner pipeline.</b> The main process is controlled by metaminer.py (yellow block). Where there are more than one input sequence/ spectrum file, the run_parallel() function is called for the organisation of simultaneous processing of those input data (orange blocks). In the the processing of sequence files, funcitons are organised in separate script depending on the format of the input file. The major sequential stages of the MetaMiner pipeline are sequence processing, spectra processing and genome analysis (green blocks).</h6>
								<h3>3.2	NeuRiPP</h3>
								<h3>3.2.1	Positive NeuRiPP predictions differ when classified by different neural network architectures</h3>
								<p>
									The same set of sequences identified by Prodigal was classified by NeuRiPP adopting three architectures. The linear CNN + lstm architecture classified more sequences as positive than just linear CNN across all classes of RiPP gene clusters. The parallel CNN classified similar numbers of sequences as positive to linear CNN + lstm for most classes except for prosteusin, microviridin, Linear azol(in)e-containing peptides (LAP), Radical S-adenosylmethionine (RaS) and bottromycin (Fig. 7). The positive predictions made by the linear CNN exhibit the least variation among all classes, with all values within +/- 2 standard deviation(391 +/- 127)(Tab S1). Notably, while the original analysis in the NeuRiPP paper reported that NeuRiPP classified near 90% of lanthipeptides as PP, there were only 2/1229 sequences classified as PP in this analysis.
								</p>
								<div id="observablehq-key-2d8ea52a"></div>
								<div id="observablehq-chart-2d8ea52a"></div>
								<h6><b>Figure 7. Class distribution of positive predictions made by NeuRiPP. </b>Three neural networks are presented in different colours. The neural network architectures used in this analysis are linear CNN (orange), parallel CNN (yellow) and linear CNN + lstm (green). (Hover over the bars and stay to see data entries.)
								</h6>

								<h3>3.2.2	HMM hit rate in NeuRiPP positive and negative subsets + breakdown of NeuRiPP positive and HMM hit by class</h3>
								<p>
									The NeuRiPP sequences were subjected to HMMscan to assess their similarities to existing PPs. Results demonstrated that more sequences generated HMM hits than those that did not in both NeuRiPP positive and negative groups. While there were more NeuRiPP positives when linear and parallel CNN were used, linear CNN + lstm predicted more sequences as NeuRiPP negative (Fig. 8). Moreover, parallel CNN predicted the highest number of NeuRiPP positives that are also HMM hits.</p>
								<p>
									Figure 9 shows the breakdown of NeuRiPP positive + HMM hit by class. Results classified by linear CNN + lstm and linear CNN exhibit a more similar distribution. 
								</p>
								<h5>Linear CNN</h5>
								<div id="observablehq-chart-f08b0783"></div>
								<h5>Linear CNN + lstm</h5>
								<div id="observablehq-chart1-f08b0783"></div>
								<h5>Parallel CNN</h5>
								<div id="observablehq-chart2-f08b0783"></div>
								<div class="inner" id="observablehq-key-f08b0783"></div>
								<p></p>
								<h6><b>Figure 8. HMM results of NeuRiPP predictions.</b> Results are divided into four categories: NeuRiPP positive (red), NeuRiPP positive + HMM hit (orange), NeuRiPP negative (green) and NeuRiPP negative + HMM hit (blue). (Hover over the segment and stay to see data entries.)
								</h6>

								<div id="observablehq-chart-efa7a84a"></div>
								<p></p>
								<div id="observablehq-key-efa7a84a"></div>
								<p>
						
								</p>
								<h6><b>Figure 9. Class breakdown of NeuRiPP positive + HMM hit results.</b> The percentage of each class is represented colours in the gradient palette. (Hover over the segment and stay to see data entries.)
								</h6>

							</div>
							


							

							
							
						</section>
						</section>
						<section class="wrapper style5">
							<ul class="actions special">
								<li><a href="methods.html" class="button primary fit">Methods</a></li>
								<li><a href="discussion.html" class="button primary fit">Discussion</a></li>
							</ul>
						</section>
					</article>

				<!-- Footer -->
					<footer id="footer">
						<ul class="icons">
							<li><a href="https://www.imperial.ac.uk/life-sciences/postgraduate/masters-courses/msc-in-bioinformatics-and-theoretical-systems-biology/" target="_blank" class="icon brands fa-dribbble"><span class="label">Dribbble</span></a></li>
							<li><a href="mailto: yuting.tseng21@imperial.ac.uk" target="_blank" class="icon solid fa-envelope"><span class="label">Email</span></a></li>
						</ul>
						<ul class="copyright">
							<li>YTT16</li><li>Design: <a href="http://html5up.net">HTML5 UP</a></li>
						</ul>
					</footer>

			</div>

		<!-- Scripts -->
			<script src="assets/js/jquery.min.js"></script>
			<script src="assets/js/jquery.scrollex.min.js"></script>
			<script src="assets/js/jquery.scrolly.min.js"></script>
			<script src="assets/js/browser.min.js"></script>
			<script src="assets/js/breakpoints.min.js"></script>
			<script src="assets/js/util.js"></script>
			<script src="assets/js/main.js"></script>
			<script type="module">
				import {Runtime, Inspector} from "https://cdn.jsdelivr.net/npm/@observablehq/runtime@4/dist/runtime.js";
				import define from "https://api.observablehq.com/@michating5511/pie-chart-for-hmm-hit.js?v=3";
				new Runtime().module(define, name => {
					if (name === "chart") return new Inspector(document.querySelector("#observablehq-chart-f08b0783"));
					if (name === "chart1") return new Inspector(document.querySelector("#observablehq-chart1-f08b0783"));
					if (name === "chart2") return new Inspector(document.querySelector("#observablehq-chart2-f08b0783"));
					if (name === "key") return new Inspector(document.querySelector("#observablehq-key-f08b0783"));
					return ["key1"].includes(name);
				});
			</script>
			<script type="module">
				import {Runtime, Inspector} from "https://cdn.jsdelivr.net/npm/@observablehq/runtime@4/dist/runtime.js";
				import define from "https://api.observablehq.com/@michating5511/first-grouped-bar-chart.js?v=3";
				new Runtime().module(define, name => {
					if (name === "chart") return new Inspector(document.querySelector("#observablehq-chart-2d8ea52a"));
					if (name === "key") return new Inspector(document.querySelector("#observablehq-key-2d8ea52a"));
				});
			</script>
			<script type="module">
				import {Runtime, Inspector} from "https://cdn.jsdelivr.net/npm/@observablehq/runtime@4/dist/runtime.js";
				import define from "https://api.observablehq.com/@michating5511/metaminer.js?v=3";
				new Runtime().module(define, name => {
				if (name === "meta_results_table") return new Inspector(document.querySelector("#observablehq-meta_results_table-a702b4ec"));
				if (name === "viewof search") return new Inspector(document.querySelector("#observablehq-viewof-search-a702b4ec"));
				});
			</script>
			<script type="module">
				import {Runtime, Inspector} from "https://cdn.jsdelivr.net/npm/@observablehq/runtime@4/dist/runtime.js";
				import define from "https://api.observablehq.com/@michating5511/positive-hit-class-breakdown.js?v=3";
				new Runtime().module(define, name => {
				if (name === "chart") return new Inspector(document.querySelector("#observablehq-chart-efa7a84a"));
				if (name === "key") return new Inspector(document.querySelector("#observablehq-key-efa7a84a"));
				});
				</script>

	</body>
</html>